### PR TITLE
[FEAT] Game tutorial/onboarding

### PR DESCRIPTION
**Pull request by Rofrtd**
# Description

This PR will implements a game onboarding experience using Reactour.

This PR is related to issue #140 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Manual testing.

https://user-images.githubusercontent.com/25412194/154172681-126c59ee-6514-417d-8229-92c9e6d36895.mp4

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/game/GameProvider.tsx [hunk 1]
```diff
@@ -10,6 +10,12 @@ import { cacheShortcuts, getShortcuts } from "features/hud/lib/shortcuts";
 
 import { startGame, MachineInterpreter } from "./lib/gameMachine";
 import { InventoryItemName } from "./types/game";
+import { inspect } from "@xstate/inspect";
+
+inspect({
```
Comment (adamhannigan) [inline src/features/game/GameProvider.tsx:15]: Should we put some dev flags around this or is this something we want to run in production?

**Review by adamhannigan — COMMENTED** [review]
LGTM mate. Just a couple WIP questions.

Is it possible to deduce that someone just created a farm without using local storage?

File: src/features/game/GameProvider.tsx (hunk 1 above)
Comment (Rofrtd) [reply src/features/game/GameProvider.tsx:15]: This was something I was testing with Spencer, but didn't quite work..
I'll remove it.

**Review by adamhannigan — COMMENTED** [review]
Looks good to me mate 💪 My only suggestion would be to use a enum to make it more readable and robust.

File: package.json [hunk 2]
```diff
@@ -16,6 +16,8 @@
     "not op_mini all"
   ],
   "dependencies": {
+    "@emotion/react": "^11.7.1",
```
[the remark this answers is not in the document]
Comment (Rofrtd) [reply package.json:20]: This is a peer dependency for reactour.

File: src/features/crops/components/Field.tsx [hunk 3]
```diff
@@ -56,6 +60,10 @@ export const Field: React.FC<Props> = ({
       return;
     }
 
+    if (onboarding && tourIsOpen) {
+      currentTourStep === 0 ? setCurrentTourStep(1) : setCurrentTourStep(8)
```
[the remark this answers is not in the document]
Comment (Rofrtd) [reply src/features/crops/components/Field.tsx]: Thanks mate, just implemented the enum.

**Review by adamhannigan — COMMENTED** [review]
Great work mate 👏 

I found one small issue though, after suggesting to 'Sync on Chain' I did that and then the last step popped up again.

1. Step asks to sync on chain
2. I click sync on chain
3. Modal opens and I accept transaction
4. Modal closes
5. Step asks to sync on chain again

Screen Shot 2022-02-18 at 9 45 30 am

I would suggest actually omitting the sync on chain part. This might not leave the best first impression for a user if they need to spend another `0.1` MATIC to save the game instantly.

When they run out of seeds, they will be prompted to sync on chain. That is probably the best time to walk them through that process

**Review by adamhannigan — APPROVED** [review]
Great work mate 🚀 This will help so many new players 🤗

**Pull request merged**